### ISSUE TITLE
Missing unique / primary key on the table - `sys_language_l10nmgr_language_restricted_record_mm`

### ISSUE THREAD

**Issue by 0xff8** [CodersCare/l10nmgr#41]
Hi everyone,

There is a table `sys_language_l10nmgr_language_restricted_record_mm` without a primary/unique key and such tables are not supported by mysql group replication. Could I ask you, devs, add/change (whatever you think is better) the table definition to create missing unique index?  

currently, the table structure is:
```SQL
CREATE TABLE `sys_language_l10nmgr_language_restricted_record_mm` (
  `uid_local` int(11) NOT NULL DEFAULT '0',
  `uid_foreign` int(11) NOT NULL DEFAULT '0',
  `tablenames` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `fieldname` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `sorting` int(11) NOT NULL DEFAULT '0',
  `sorting_foreign` int(11) NOT NULL DEFAULT '0',
  KEY `uid_local_foreign` (`uid_local`,`uid_foreign`),
  KEY `uid_foreign_tablefield` (`uid_foreign`,`tablenames`(40),`fieldname`(3),`sorting_foreign`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

**Comment by Bunnyfield** [comment]
This seems to be a problem of any usual MM table in TYPO3 then, since i.e. `sys_category_record_mm` does not contain a unique key either. We could provide a primary key for the l10nmgr table, but first need to check if this had any impact on how TYPO3 deals with those records.

**Comment by 0xff8** [comment]
Hi @Bunnyfield ,
Thank for the response.   
I confirm that typo3 table `sys_category_record_mm` has no unique index. I usually add the missing index by creating it in one of project-related extensions. Do you suggest to do the same for `sys_language_l10nmgr_language_restricted_record_mm` (if so, what fields should be included in the unique index) ? 

I also know , that `sys_category_record_mm` in typo3 version 12 has a unique index. So, I thought if you are planning to support typo3 v12, may be you could add this index directly in the `ext_tables.sql` and have it fixed for all versions?

**Comment by Bunnyfield** [comment]
So it's not about adding a primary key field, but a unique index would be enough?

**Comment by 0xff8** [comment]
Yes, that shall be enought for mysql group replicaiton. If you could do so , that would be amazing :pray:

**Comment by Bunnyfield** [comment]
Sorry for the late reply, but actually those keys should have been created by the core automagically. At the same time, the restriction handling has been moved from MM tables to itemsProcFunc and SiteLanguage. So I will close this issue,
